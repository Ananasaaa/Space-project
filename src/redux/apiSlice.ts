import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

export const nasaApi = createApi({
  reducerPath: 'nasaApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'https://api.nasa.gov/',
  }),
  endpoints: (builder) => ({
    getSpaceImage: builder.query({
      query: () => ({
        url: 'planetary/apod',
        params: {
          api_key: process.env.REACT_APP_NASA_KEY as string,
        },
      }),
    }),
    getMarsPhotos: builder.query({
      query: ({ sol = 4000, camera, page = 1 }) => ({
        url: 'mars-photos/api/v1/rovers/curiosity/photos',
        params: {
          sol,
          page,
          ...(camera ? { camera } : {}),
          api_key: process.env.REACT_APP_NASA_KEY as string,
        },
      }),
    }),
  }),
});

export const { useGetSpaceImageQuery, useGetMarsPhotosQuery } = nasaApi;
