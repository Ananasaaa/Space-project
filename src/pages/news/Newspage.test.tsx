import { render, screen } from '@testing-library/react';
import Newspage from './index';
import { useGetNasaNotificationsQuery } from '../../redux/apiSlice';

jest.mock('../../redux/apiSlice');

describe('Newspage component', () => {
  it('should render notifications correctly', () => {
    (useGetNasaNotificationsQuery as jest.Mock).mockReturnValue({
      data: [
        {
          messageID: '1',
          messageType: 'Test Type',
          messageIssueTime: '2024-04-24T12:00:00Z',
          messageBody:
            'Summary: Test summary text ## Notes: Notes: Some additional notes',
          messageURL: 'https://example.com/full-message',
        },
      ],
      isLoading: false,
      isError: false,
    });

    render(<Newspage />);

    expect(screen.getByText('NASA Notifications')).toBeInTheDocument();
    expect(screen.getByText('Test Type')).toBeInTheDocument();
    expect(screen.getByText('Test summary text')).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /View Full Message/i })
    ).toBeInTheDocument();
  });
});
