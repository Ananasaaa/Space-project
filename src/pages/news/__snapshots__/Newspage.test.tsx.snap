// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newspage component should render notifications correctly 1`] = `
<DocumentFragment>
  <div
    class="notifications-page"
    style="background-image: url(/img/bg_newspage.jpg); background-size: cover; background-position: right center; background-repeat: no-repeat;"
  >
    <h1
      class="page-title"
    >
      NASA Notifications
    </h1>
    <div
      class="notifications-list"
    >
      <div
        class="notification-card fade-in"
      >
        <h2
          class="message-type"
        >
          🛰️ Test Type
        </h2>
        <p
          class="message-time"
        >
          📅 4/24/2024, 3:00:00 PM
        </p>
        <p
          class="message-summary"
        >
          📌 Test summary text
        </p>
        <p
          class="message-notes"
        >
          🗒️ Notes: Some additional notes
        </p>
        <a
          class="message-link"
          href="https://example.com/full-message"
          rel="noopener noreferrer"
          target="_blank"
        >
          🔗 View Full Message
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
